src/hdf5_interface.F90

```diff
@@ -1529,11 +1531,94 @@ end subroutine write_string
 
   subroutine read_string(group_id, name, buffer, indep)
     integer(HID_T), intent(in) :: group_id
+    character(*), intent(in)            :: name    ! name for data
+    character(*), intent(inout), target :: buffer  ! read data to here
+    logical,      intent(in), optional  :: indep ! independent I/O
+
+    integer :: hdf5_err
+    integer :: data_xfer_mode
+#ifdef PHDF5
+    integer(HID_T) :: plist   ! property list
+#endif
+    integer(HID_T) :: dset    ! data set handle
+    integer(HID_T) :: dspace  ! data or file space handle
+    integer(HID_T) :: filetype
+    integer(HID_T) :: memtype
+    integer(HSIZE_T) :: size
+    integer(HSIZE_T) :: n
+    type(c_ptr) :: f_ptr
+
+    ! Set up collective vs. independent I/O
+    data_xfer_mode = H5FD_MPIO_COLLECTIVE_F
+    if (present(indep)) then
+      if (indep) data_xfer_mode = H5FD_MPIO_INDEPENDENT_F
+    end if
+
+    ! Get dataset and dataspace
+    call h5dopen_f(group_id, trim(name), dset, hdf5_err)
+    call h5dget_space_f(dset, dspace, hdf5_err)
+
+    ! Make sure buffer is large enough
+    call h5dget_type_f(dset, filetype, hdf5_err)
+    call h5tget_size_f(filetype, size, hdf5_err)
+    if (size > len(buffer) + 1) then
+      call fatal_error("Character buffer is not long enough to &
+           &read HDF5 string.")
+    end if
+
+    ! Get datatype in memory based on Fortran character
+    n = len(buffer)
+    call h5tcopy_f(H5T_FORTRAN_S1, memtype, hdf5_err)
+    call h5tset_size_f(memtype, n, hdf5_err)
+
+    ! Get pointer to start of string
+    f_ptr = c_loc(buffer(1:1))
+
+    if (using_mpio_device(group_id)) then
+#ifdef PHDF5
```

Comment:
Not quite. If you're not using PHDF5, when the file is opened, the file driver will just be the default one which uses POSIX file-system functions rather than the MPIO driver which relies on MPI-IO for accessing files. So, if only MPI, `using_mpio_devce(group_id)` will be false.